newm-server/src/main/kotlin/io/newm/server/features/marketplace/repo/MarketplaceRepositoryImpl.kt

```diff
@@ -14,33 +28,50 @@ import io.newm.server.features.marketplace.parser.parseQueue
 import io.newm.server.features.marketplace.parser.parseSale
 import io.newm.server.features.song.database.SongTable
 import io.newm.server.ktx.getSecureConfigString
+import io.newm.shared.exception.HttpPaymentRequiredException
+import io.newm.shared.exception.HttpUnprocessableEntityException
 import io.newm.shared.koin.inject
+import io.newm.shared.ktx.coLazy
 import io.newm.shared.ktx.debug
 import io.newm.shared.ktx.epochSecondsToLocalDateTime
+import io.newm.shared.ktx.getConfigLong
 import io.newm.shared.ktx.info
 import io.newm.shared.ktx.orZero
 import io.newm.shared.ktx.warn
-import kotlinx.coroutines.runBlocking
+import io.newm.txbuilder.ktx.mergeAmounts
+import io.newm.txbuilder.ktx.toCborObject
+import io.newm.txbuilder.ktx.toNativeAssetCborMap
 import org.jetbrains.exposed.dao.id.EntityID
 import org.jetbrains.exposed.sql.and
 import org.jetbrains.exposed.sql.transactions.transaction
 import org.koin.core.parameter.parametersOf
 import org.slf4j.Logger
+import java.math.BigInteger
 import java.util.UUID
 
 private const val SALE_TIP_KEY = "saleTip"
 private const val QUEUE_TIP_KEY = "queueTip"
 
 internal class MarketplaceRepositoryImpl(
     private val environment: ApplicationEnvironment,
+    private val configRepository: ConfigRepository,
     private val cardanoRepository: CardanoRepository
 ) : MarketplaceRepository {
     private val log: Logger by inject { parametersOf(javaClass.simpleName) }
-
-    private val pointerPolicyId: String by lazy {
-        runBlocking {
-            environment.getSecureConfigString("marketplace.pointerPolicyId")
-        }
+    private val queueContractAddress: String by coLazy {
+        environment.getSecureConfigString("marketplace.queue.contractAddress")
```

Comment:
These should not be environment strings. They should come from the config table. If they are an environment string, the whole server needs to be reloaded if we need to change the smart contracts, etc... If we just put them as configured from the db, we can update all of these values on the fly.

None of these values need to be encrypted or "secure". They are public on the blockchain anyway.